### PR TITLE
Add GHC 8.10.5 support

### PR DESCRIPTION
**Pull request by Ailrun**
This PR
- ~~drops 8.10.2 support~~ (We need to wait until 8.10.6)
- adds 8.10.5 support

~~Resolves~~ Related to #1898, ~~probably resolves~~ related to #1465

~~CI fails until https://github.com/hsyl20/ghc-api-compat/pull/11 is resolved (If it takes too long, I will redirect the dependency to PR head)~~ I redirected the dependency

### CODE REVIEW

File: .github/workflows/build.yml
Comment (Ailrun) [inline .github/workflows/build.yml:84]: Which part was the normalization? Was it `GHC_VER=$(echo $GHC_VER | sed 's/8.10.2.2/8.10.2/g')` part?

Comment (jneira) [reply .github/workflows/build.yml:84]: yep, not sure if 8.10.5 is in the choco package yet 🤔

Comment (Ailrun) [reply .github/workflows/build.yml:84]: @jneira it's unfortunately not in the list... I added a comment on the package

Comment (jneira) [reply .github/workflows/build.yml:84]: it still is not

Comment (Ailrun) [reply .github/workflows/build.yml:84]: I also sent a mail to the maintainer. I hope this will be resolved soon...

Comment (jneira) [reply .github/workflows/build.yml:84]: There is a [github repo](https://github.com/Mistuke/GhcChoco), i usually open issues about there

Comment (Ailrun) [reply .github/workflows/build.yml:84]: Oh I didn't know that, I will make a PR

Comment (Ailrun) [reply .github/workflows/build.yml:84]: @jneira Actually the maintainer was faster than my PR :) Now there's GHC 8.10.5

**Comment by Ailrun** [comment]
Maybe blocked by upstream? https://gitlab.haskell.org/ghc/ghc/-/issues/19763#note_357220

File: .github/workflows/test.yml
Comment (Ailrun) [inline .github/workflows/test.yml:47]: Windows 8.10.4 should not be removed

**Comment by jneira** [comment]
We should estimate the impact of the bug, but i suppose we will have to add support for 8.10.5, as people is gonna use it anyways :face_with_head_bandage: 
It would suppose mark tests as broken for ghc 8.10.5, it will let us make progress and help us to estimate the impact at same time.

**Comment by jneira** [comment]
To try to avoid this scenario i think we should build and test hls against ghc branches that will serve as base for minor releases (8.10 and 9.0). This will help ghc developers to catch them earlier too.

**Comment by Ailrun** [comment]
@jneira It looks like that GHC plugin is the only case that causes the problem in our test set, so I think it's not the worst situation.

**Comment by Ailrun** [comment]
I don't know why but MacOS build stucks after completed Brittany install...

**Comment by konn** [comment]
It gets stuck also on my laptop (MacBook Pro), preprocessing the `network` package forever...

**Comment by Ailrun** [comment]
@konn so GHC 8.10.5 on Mac is kinda... broken? (as it does not work with both Cabal and Stack)

**Comment by konn** [comment]
@Ailrun I haven't done any deep investigation, but it seems there is a problem, I'm afraid...

**Comment by Ailrun** [comment]
@konn Could you check whether the network package alone causes the same issue? If so we can report a more specific issue on GHC

**Comment by konn** [comment]
@Ailrun Ah, I've just confirmed that building the network package alone causes the same issue, and reported it to GHC GitLab: https://gitlab.haskell.org/ghc/ghc/-/issues/19968.

**Comment by Ailrun** [comment]
Awesome, thank you for the fast process!

**Comment by Ailrun** [comment]
Should we leave 8.10.2 until GHC 8.10.6 is out? Especially for MacOS... I disabled MacOS CI and build, and I don't know it's possible to generate a binary in MacOS with GHC 8.10.5

**Comment by jneira** [comment]
> Should we leave 8.10.2 until GHC 8.10.6 is out

I think we should be cautious and keep it some time more

**Review by jneira — APPROVED** [review]
I think we could merge this to make it available to user, which are being used it already.
Great work as usual, many thanks 🙂

**Comment by Ailrun** [comment]
@jneira Yes, now we should change the required test to Windows 8.10.5 with test and Windows 8.10.4 without test. I will change them.

**Pull request merged**
